Liver fixes

DaxDupont opened the pull request:
[Changelogs]: 
:cl: Dax Dupont & Naksu
fix: Toxin loving species won't take liver damage from toxins.
fix: Liver now gets 0.01 damage per unit of toxin subtype reagent every 2 seconds instead of 0.5. IE 30 units of whatever won't kill you within 10 seconds anymore.
fix: Moved peacekeeper borg reagents to other_reagents so they don't do liver damage and they aren't really toxins
refactor: Renamed borg synthpax path to be more inline and put them next to the other borg reagents.
/:cl:

[why]: These might be balance instead, but I doubt killing people in 10 seconds because they have 30u of mutetoxin was intended.
Fixes #37171
Fixes #37164
Closes #37170

DaxDupont committed:
Rebalances lethality and adds a traitcheck

DaxDupont committed:
Moves borg reagent to other_reagents since these shouldn't do damage at all and are hardly toxins

vuonojenmustaturska commented:
looks alright at a glance

DaxDupont committed:
Lower it more because this is probably gonna hurt a lot real fast

81Denton commented:
Why base it off toxin subtype reagents instead of toxloss/towpwr? 
There are plenty of nonlethal toxins like mute toxin, mime's bane, mint toxin, tirizine, mindbreaker etc.

DaxDupont commented:
Because tox loss is the current amount of toxins lost and that doesn't go down on its own much. reagents however get removed if they do tox damage.

Toxpwr is set to 0 for a lot of reagents that do random tox effects and what not.

Toxins are still toxins even if non-lethal I guess. If you want to split them off or add a harmless var or something go ahead, but this is my solution for now.

vuonojenmustaturska commented:
Design-wise it should be fine for things that are not intended to kill you still present some effects that you might not enjoy.

Also, coming up next, onions are toxic to catgirls

DaxDupont committed:
Naku's suggestion

derp

Thanks cobby

readds this check

almost nerfed cybernetic livers oops

Aaaaaa

ExcessiveUseOfCobblestone commented on code/modules/surgery/organs/liver.dm:
```diff
@@ -25,22 +25,17 @@
-					for(var/I in C.reagents.reagent_list)
-						var/datum/reagent/pickedreagent = I
-						if(is_type_in_typecache(pickedreagent, toxinstypecache))
+						var/thisamount = C.reagents.get_reagent_amount(initial(pickedreagent.id))
```
is pickedreagent a typepath? looks like an instance from `C.reagents.reagent_list`

vuonojenmustaturska replied:
it's an instance but `initial(thing.id)` is faster for instance-invariant data that has to be read from the type

ExcessiveUseOfCobblestone replied:
I hate byond...

ExcessiveUseOfCobblestone commented on code/modules/surgery/organs/liver.dm:
```diff
@@ -25,22 +25,17 @@
-					damage += toxamount*toxLethality
-
+						else
+							damage += min((thisamount <= toxTolerance ? 0 : (thisamount*toxLethality)),0)
```
It's always going to not do any damage?

ExcessiveUseOfCobblestone replied:
Also please explain math like this with commenting so idiots like myself don't break anything in the future :)

ExcessiveUseOfCobblestone commented:
@DaxDupont please explain Line 38.

DaxDupont commented:
@ExcessiveUseOfCobblestone why, it's just a ternary operator

DaxDupont commented:
It's basically 
```
if (thisamount <= toxTolerance)
 damage += 0
else 
 damage += thisamount*toxLethality
```

ExcessiveUseOfCobblestone reviewed (changes requested):
I hate ~~math~~ ternary operators...

ExcessiveUseOfCobblestone commented:
@DaxDupont i've used them before, I should have been more explicit in that I was confused why you were checking for the same static number multiple times.

DaxDupont commented on code/modules/surgery/organs/liver.dm:
```diff
@@ -25,22 +25,17 @@
-					damage += toxamount*toxLethality
-
+						else
+							damage += max((thisamount <= toxTolerance ? 0 : (thisamount*toxLethality)), 0)
```
Actually you're right.

ExcessiveUseOfCobblestone replied:
Because I have corrected you, you now have to code archaeology for me to enjoy. I think that's a fair price to pay.

DaxDupont replied:
pls no

ExcessiveUseOfCobblestone commented:
No problem but you still owe me archaeology now hehe

ExcessiveUseOfCobblestone commented on code/modules/surgery/organs/liver.dm:
```diff
@@ -25,22 +25,17 @@
-						if(is_type_in_typecache(pickedreagent, toxinstypecache))
+						var/thisamount = C.reagents.get_reagent_amount(initial(pickedreagent.id))
+						if (thisamount <= toxTolerance && thisamount)
 							C.reagents.remove_reagent(initial(pickedreagent.id), 1)
```
I know you didn't touch this but why is this 1 and not normal metabolism of the reagent?

ExcessiveUseOfCobblestone reviewed (approved):
In the future we should remove the toxin check from livers and have them based off a toxicity variable as I think @81Denton mentioned.

Although for the purpose of some medicines being bad for your liver as well :p

DaxDupont commented:
Apparently reagent amount can be negative very briefly and naksu told me to readd the check.

DaxDupont commented on code/modules/surgery/organs/liver.dm:
```diff
@@ -25,22 +25,17 @@
-						var/datum/reagent/pickedreagent = I
-						if(is_type_in_typecache(pickedreagent, toxinstypecache))
+						var/thisamount = C.reagents.get_reagent_amount(initial(pickedreagent.id))
+						if (thisamount <= toxTolerance && thisamount > 0)
```
apparently not according to naksu

ExcessiveUseOfCobblestone commented:
`get_reagent_amount` should be checking that and returning 0 I think, not every local incident where it might be negative.

DaxDupont commented:
It should but it doesn't if volume is negative somehow.
Don't want to touch that right now. If someone else wants to touch it go ahead but I am just listening to the maintainer for now since I might have to go soon and this shit is causing too much death and I have to go in the next hour and this will probably be speedmerged so I am gonna leave it as is.

davethwave commented:
Instead of .01 how about .03 or something like that? Might give a actual use to robotic organs that you can make in medical.

DaxDupont commented:
@davethwave that's why we're test merging this to check for the right values.

DaxDupont commented:
Since #37179 exists i am gonna remove the check here.

davethwave commented:
How about amatin? Since the toxin has a delayed effect, shouldn't it not deal damage until it activates and do a stacked amount of damage to the liver as well as the basic tox damage it does?

DaxDupont commented:
@davethwave feel free to add it after or something, but it liver damage isn't direct damage unless the liver is gone. Regular toxinloss doesn't affect the liver itself. Delayed toxins are still toxins in the system, they just take a while to take damage.

We can shit up the liver with snowflake code but in the current state people are dying almost instantly with any toxins and this fixes that. If you want to add more features or snowflake memes please do so after 20-30u's of Mime's Bane and other memes like that stop killing people.

vuonojenmustaturska commented:
fastmerging this with kor's blessing
